configuration/config-txt/memory.md

```diff
@@ -2,9 +2,13 @@
 
 ## gpu_mem
 
-GPU memory in megabytes, sets the memory split between the CPU and GPU; the CPU gets the remaining memory. The minimum value is `16`; the maximum value is `192`, `448`, or `944`, depending on whether you are using a 256MB, 512MB, or 1024MB Pi. The default value is `64`. For the Raspberry Pi 4, which is available in 1GB, 2GB and 4GB versions, the minimum and maximum value are the same as for a 1GB device.
+GPU memory in megabytes, sets the memory split between the CPU and GPU; the CPU gets the remaining memory. The minimum value is `16`; the maximum value is `192`, `448`, or `944`, depending on whether you are using a 256MB, 512MB, or 1024MB Pi. The default value is `64`. For the Raspberry Pi 4, which is available in 1GB, 2GB and 4GB versions, the minimum and maximum values are the same as for a 1GB device.
 
-Setting `gpu_mem` to low values may automatically disable certain firmware features, as there are some things the GPU cannot do if it has access to too little memory. So if a feature you are trying to use isn't working, try setting a larger GPU memory split.
+`gpu_mem` refers to memory that is addressable from the GPU, which includes the VPU, HVS, legacy codecs (e.g. H264), and camera, and on devices before the Raspberry Pi 4, the 3D system. The Raspberry Pi 4 3D system has it's own Memory Mangement Unit (MMU) so textures and other GL resources are not allocated from the `gpu_mem` but Linux system memory instead. This means that `gpu_mem` can be set to a lower value, so even if you are using the H264 and camera then 128MB will probably be enough. On earlier models without the 3D MMU, you may need up to 256 or 512 in some more unusual cases. 
+
+For performance reasons, you should set `gpu_mem` as low as possible to give the Linux system as much memory as possible. However, setting `gpu_mem` to too low values may automatically disable certain firmware features, as there are some things the GPU cannot do if it has access to too little memory. So if a feature you are trying to use isn't working, try setting a larger GPU memory split.
+
+Values of `gpu_mem` over 512 are not recommended, will provide no performance improvements, and are untested.
```

Comment:
To cover @andrum99 's remarks, probably the above sentence could be placed into the first paragraph, right after "`depending on whether you are using a 256MB, 512MB, or 1024MB Pi.`"
And as `944` is a technical max value only, it could be named like this. So the whole part could be then:

> The minimum value is `16`; the technical maximum value is `192`, `448`, or `944`, depending on whether you are using a 256MB, 512MB, or 1024MB Pi. But values of `gpu_mem` over 512 are not recommended, will provide no performance improvements, and are untested.

Reply:
This is probably a decision for the copy-editors.

Reply:
It is also the case that values of gpu_mem that approach the total size of physical RAM in the system are problematic. In other words, if gpu_mem is set too close to the size of RAM then bad things will happen. The docs don't mention this at the moment

Reply:
@andrum99 
However there is a maximum allowed value that assures that a minimum memory is left for the system. I think otherwise it should be clear that more gpu_mem means less system RAM, already by the word "share". If one touches these values, one should be aware that a reasonable share between system and GPU should be chosen, that fits to the needs and software that runs.

Reply:
I'd added an warning in the paragraph but have retained the previous warning. Hopefully people won't miss both of them.